testsuite/integration-tests-spring/deployment/src/test/java/org/jboss/resteasy/test/spring/deployment/AddSpringResteasyAsResourceRootTest.java

```diff
@@ -77,6 +82,16 @@ private static WebArchive createDeploymentWithSpringMvcDispatcher() {
                 .addAsWebInfResource(AddSpringResteasyAsResourceRootTest.class.getPackage(), "mvc-dispatcher-servlet/web.xml", "web.xml")
                 .addAsWebInfResource(AddSpringResteasyAsResourceRootTest.class.getPackage(), "mvc-dispatcher-servlet/mvc-dispatcher-servlet.xml", "mvc-dispatcher-servlet.xml")
                 .addAsWebInfResource(AddSpringResteasyAsResourceRootTest.class.getPackage(), "mvc-dispatcher-servlet/applicationContext.xml", "applicationContext.xml");
+
+        archive.addAsManifestResource(PermissionUtil.createPermissionsXmlAsset(
+            new PropertyPermission("arquillian.*", "read"),
+            new ReflectPermission("suppressAccessChecks"),
+            new RuntimePermission("accessDeclaredMembers"),
+            new RuntimePermission("getClassLoader"),
+            new FilePermission("<<ALL FILES>>", "read"),
+            new LoggingPermission("control", "")
+        ), "permissions.xml");
```

Comment:
I can not reproduce the issue of needing to add permission "insertProvider".
This permission block has a comment, "remaining permissions needed to run springframework"